### PR TITLE
feat(server): add DELETE session API 

### LINKED ISSUE #11665: 🗺️ [sdk/connectors] Session APIs
SDK and API support for grouping traces into user sessions

## Spike

- [ ]

## Front End

- [ ]

## Back End

- [ ] #11692

## Open Questions

-

### PR DESCRIPTION
**Pull request by mikeldking**
## Summary

Closes #11665

- Add `DELETE /v1/sessions/{session_identifier}` endpoint (single session delete by GlobalID or session_id, returns 204/404)
- Add `POST /v1/sessions/delete` endpoint (bulk delete by session_id list, silently skips missing, returns 204/422)
- Add `Sessions.delete()` and `AsyncSessions.delete()` methods to Python client
- Add `deleteSession()` function to TypeScript client
- Regenerate OpenAPI schema and client types
- Add 7 server unit tests covering delete by session_id, GlobalID, 404, cascade, bulk, partial match, and empty list validation

All deletions cascade through traces, spans, and annotations via existing FK constraints (no migration needed). Cache invalidation via `SpanDeleteEvent`.

## Test plan

- [x] `pytest tests/unit/server/api/routers/v1/test_sessions.py` — 21 passed (7 new + 14 existing)
- [ ] CI passes

### CODE REVIEW

**Comment by mikeldking** [comment]
## Latest changes (507de453d)

### `DeleteSessionsRequestBody` naming & GlobalID support

- **Replaced generic `RequestBody[list[str]]`** with a dedicated model using a `session_identifiers` field — the OpenAPI schema now has a meaningful field name instead of `data: list[str]` with `"title": "Data"`
- **Added dual-mode identifier support**: the bulk delete endpoint now accepts both GlobalIDs and user-provided `session_id` strings
- **Type validation**: all identifiers in a single request must be the same type (all GlobalIDs or all session_id strings); mixed input returns 422
- **Single-pass parsing**: identifiers are classified in one loop instead of three list comprehensions
- **Updated route description** to document GlobalID support
- **Updated generated client types** (`__init__.py` and `.dataclass.py`)
- **Added tests**: bulk delete by GlobalIDs, mixed identifier type rejection (422)

All 27 unit tests pass.

**Comment by mikeldking** [comment]
## Latest changes (d75f437a5)

### OpenAPI schema + client SDK support for bulk delete

- **Regenerated `schemas/openapi.json`** — `DeleteSessionsRequestBody` now shows `session_identifiers` field with description instead of generic `data`
- **Regenerated client types** — Python (`__init__.py`, `.dataclass.py`) and TypeScript (`v1.ts`) all updated from schema
- **Python client**: added `Sessions.bulk_delete()` and `AsyncSessions.bulk_delete()` with `session_ids` parameter
- **TypeScript client**: added `deleteSessions()` function and exported from `sessions/index.ts`
- **Tests**: added sync/async client unit tests for `bulk_delete` (request body shape, empty list validation, HTTP error handling)

All 47 tests pass (27 server + 20 client).

File: packages/phoenix-client/src/phoenix/client/__generated__/v1/.dataclass.py [hunk 1]
```diff
@@ -0,0 +1,1360 @@
+# generated by datamodel-codegen:
```
Comment (mikeldking) [inline packages/phoenix-client/src/phoenix/client/__generated__/v1/.dataclass.py]: @claude why is this file not gitignored?
